pyqtgraph/opengl/items/GLTextItem.py

```diff
@@ -0,0 +1,92 @@
+from typing import List, Tuple, Union
+from OpenGL.GL import *
+from OpenGL.GLUT import *
+import numpy as np
+import pyqtgraph.functions as fn
+from pyqtgraph.opengl.GLGraphicsItem import GLGraphicsItem
+
+class GLTextItem(GLGraphicsItem):
+    """Draws text in 3D."""
+
+    def __init__(self, **kwds):
+        """All keyword arguments are passed to setData()"""
+        GLGraphicsItem.__init__(self)
+        glopts = kwds.pop('glOptions', 'additive')
+        self.setGLOptions(glopts)
+
+        self.pos = np.array([0.0, 0.0, 0.0])
+        self.color = (1.0, 1.0, 1.0, 1.0)
+        self.text = ''
+        self.font = GLUT_BITMAP_HELVETICA_18
+
+        self.setData(**kwds)
+        glutInit()
+    
+    def setData(self, **kwds):
+        """
+        Update the data displayed by this item. All arguments are optional;
+        for example it is allowed to update text while leaving colors unchanged, etc.
+
+        ====================  ==================================================
+        **Arguments:**
+        ------------------------------------------------------------------------
+        pos                   (3,) array of floats specifying text location.
+        color                 (4,) array of floats (0.0-1.0) or
+                              tuple of floats specifying
+                              a single color for the entire item.
+        text                  String to display.
+        font                  GLUT font. (Default: GLUT_BITMAP_HELVETICA_18)
+        ====================  ==================================================
+        """
+        args = ['pos', 'color', 'text', 'font']
+        for k in kwds.keys():
+            if k not in args:
+                raise Exception('Invalid keyword argument: %s (allowed arguments are %s)' % (k, str(args)))
+        for arg in args:
+            if arg in kwds:
+                value = kwds[arg]
+                if arg == 'pos':
+                    if isinstance(value, np.ndarray):
+                        if value.shape != (3,):
+                            raise Exception('"pos.shape" must be (3,).')
+                    elif isinstance(value, (tuple, list)):
+                        if len(value) != 3:
+                            raise Exception('"len(pos)" must be 3.')
+                elif arg == 'color':
+                    if isinstance(value, np.ndarray):
```

Comment:
We can probably get rid of all these checks, and let any exception that come up during `fn.mkColor` handle issues here.  For example, this would fail on the case of `color  = (127, 127, 127)` but `fn.glColor` would return a correct result for that use-case.

Reply:
I fixed in [389bce9](https://github.com/pyqtgraph/pyqtgraph/pull/1776/commits/389bce973f6dfaa44707ba402a6f63ac2b5b62b0).